code/game/mecha/mech_fabricator.dm

```diff
@@ -230,29 +221,6 @@
 		output += "\[<a href='?src=\ref[src];process_queue=1'>Process queue</a> | <a href='?src=\ref[src];clear_queue=1'>Clear queue</a>\]"
 	return output
 
-/obj/machinery/mecha_part_fabricator/proc/update_tech()
-	if(!files)
-		return
-	var/output
-	for(var/datum/tech/T in files.known_tech)
-		if(T && T.level > 1)
-			var/diff
-			switch(T.id)
-				if("materials")
-					//one materials level is 1/32, so that max level is 0.75 coefficient
-					diff = round(initial(resource_coeff_tech) - (initial(resource_coeff_tech)*(T.level-1))/32,0.01)
-					if(resource_coeff_tech>diff)
-						resource_coeff_tech = diff
-						output+="Production efficiency increased.<br>"
-				if("programming")
-					//one materials level is 1/40, so that max level is 0.8 coefficient
-					diff = round(initial(time_coeff_tech) - (initial(time_coeff_tech)*(T.level-1))/40,0.1)
-					if(time_coeff_tech>diff)
-						time_coeff_tech = diff
-						output+="Production routines updated.<br>"

```

Comment:
As I explained this to @KorPhaeron. The reason this was removed was for consistency--upgrades should be what makes things faster, not an arbitrary programming tech tree (which you can cheese higher and higher with the Experimentor). The majority of production time improvements don't come from this state anyway; the best you can hope for is a 20% improvement, and that's if you have programming level 9.